source/core/StringUtilities.cpp

```diff
@@ -71,14 +69,8 @@ Int32 SubString::DigitValue(Utf32Char codePoint, Int32 base)
     
     return value;
 }
-//------------------------------------------------------------
-Boolean SubString::Compare(const Utf8Char* begin2, IntIndex length2) const
-{
-    if (length2 != Length()) {
-        return false;
-    }
-    const Utf8Char* sCompare = _begin;
-    const Utf8Char* sEnd = _begin+length2;
+
+inline Boolean CompareString(const Utf8Char* sCompare, const Utf8Char* sEnd, const Utf8Char* begin2) {
```

Comment:
I'm not sure what's confusing about it, but I added a comment explaining the parameters, and renamed begin2 to sCompare2.

I don't like renaming sCompare to begin1 because the variable is incremented as it steps through the array, so after one iteration it's no longer the beginning.   This was also true of begin2, which I don't like.
So instead I  added '1' to the first two args, and renamed the other string pointer to sCompare2.